### ISSUE TITLE
Tox.ini should use twisted's extras

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8862]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)| @derwolfe reported|
|-|-|
|Trac ID|trac#8862|
|Type|enhancement|
|Created|2016-10-15 20:59:13Z|

Now that optional extras are supported by Twisted's `twisted/python/_setup.py`, tox should use them to install the extras instead of naming them directly in `tox.ini`.

<details><summary>Searchable metadata</summary>

```
trac-id__8862 8862
type__enhancement enhancement
reporter__derwolfe derwolfe
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1476565153396797 1476565153396797
changetime__1477905579863317 1477905579863317
version__None None
owner__derwolfe derwolfe

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_1"></a>|@derwolfe commented|
|-|-|

PR here: https://github.com/twisted/twisted/pull/565

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc set status to `closed`|
|-|-|

In changeset f1e1e8aa870aff8c2dfb3dce9cb978a32624c998
```
#!CommitTicketReference repository="" revision="f1e1e8aa870aff8c2dfb3dce9cb978a32624c998"
Merge pull request #568 from twisted/8862-tox-extras-2

Author: derwolfe
Reviewer: rodrigc
Fixes: #8862

Use extras in tox.ini
```
